server/src/main/java/io/crate/expression/eval/EvaluatingNormalizer.java

```diff
@@ -174,17 +174,19 @@ public void writeTo(StreamOutput out) throws IOException {
                         if (out.getVersion().onOrAfter(Version.V_4_2_4)) {
                             super.writeTo(out);
                         } else {
-                            io.crate.expression.predicate.MatchPredicate.INFO.writeTo(out);
+                            if (out.getVersion().before(Version.V_5_0_0)) {
+                                io.crate.expression.predicate.MatchPredicate.TEXT_MATCH.writeAsFunctionInfo(out);
+                            }
                             if (out.getVersion().onOrAfter(Version.V_4_1_0)) {
                                 Symbols.nullableToStream(filter, out);
                             }
                             Symbols.toStream(arguments, out);
                             if (out.getVersion().onOrAfter(Version.V_4_2_0)) {
-                                out.writeBoolean(signature != null);
-                                if (signature != null) {
-                                    signature.writeTo(out);
-                                    DataTypes.toStream(returnType, out);
+                                if (out.getVersion().before(Version.V_5_0_0)) {
+                                    out.writeBoolean(true);
                                 }
+                                signature.writeTo(out);
+                                DataTypes.toStream(returnType, out);
```

Comment:
Aren't some of these after conditions dead code due to the 

```
if (out.getVersion().onOrAfter(Version.V_4_2_4)) {
                            super.writeTo(out);
```

at the top?

Reply:
Actually, my added `before 5.0.0` checks can be removed due to that, good catch, thx.
But the other `onOrAfter` must stay as they match  "version <= 4.2.4 && version >= 4.1.0/4.2.0" conditions.